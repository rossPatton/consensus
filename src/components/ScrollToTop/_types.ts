import React from 'react';
import * as H from 'history';

export type tProps = {
  children?: React.ReactElement<any>,
  location: H.Location,
}
