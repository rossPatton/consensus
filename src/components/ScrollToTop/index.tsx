import React, {PureComponent} from 'react';
import { withRouter } from 'react-router-dom';

import {tProps} from './_types';

class ScrollToTopComponent extends PureComponent<tProps> {
  componentDidUpdate(prevProps: tProps) {
    if (
      this.props.location.pathname !== prevProps.location.pathname
    ) {
      window.scrollTo(0, 0);
    }
  }

  render() {
    return this.props.children || null;
  }
}

// @ts-ignore
export const ScrollToTop = withRouter(ScrollToTopComponent);
