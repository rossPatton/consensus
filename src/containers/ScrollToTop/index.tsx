import { PureComponent } from 'react';

import { withRouter } from '~app/utils';
import { tProps } from './_types';

class ScrollToTopComponent extends PureComponent<tProps> {
  componentDidUpdate(prevProps: tProps) {
    console.log("ðŸš€ ~ componentDidUpdate ~ props:", this.props)
    console.log("ðŸš€ ~ componentDidUpdate ~ prevProps:", prevProps)
    if (
      this.props.location.pathname !== prevProps.location.pathname
    ) {
      window.scrollTo(0, 0);
    }
  }

  render() {
    return this.props.children || null;
  }
}

// @ts-ignore
const ScrollToTop = withRouter(ScrollToTopComponent);
export default ScrollToTop;
