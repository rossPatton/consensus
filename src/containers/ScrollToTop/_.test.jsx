import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import {act, create} from 'react-test-renderer';

import ScrollToTop from '.';

describe('containers/ScrollToTop', () => {
  beforeAll(() => {
    global.scrollTo = jest.fn();
  });

  it('renders null', () => {
    const component = create((
      <MemoryRouter>
        <ScrollToTop location={{pathname: '/'}} />
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render children', () => {
    const component = create((
      <MemoryRouter>
        <ScrollToTop location={{pathname: '/'}}>
          test
        </ScrollToTop>
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('updates on pathname change', () => {
    const component = create((
      <MemoryRouter>
        <ScrollToTop location={{pathname: '/'}} />
      </MemoryRouter>
    ));

    act(() => {
      component.update((
        <MemoryRouter>
          <ScrollToTop location={{pathname: '/user/1'}} />
        </MemoryRouter>
      ));
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
