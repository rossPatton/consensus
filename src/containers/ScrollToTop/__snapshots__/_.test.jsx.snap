// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/ScrollToTop render children 1`] = `"test"`;

exports[`containers/ScrollToTop renders null 1`] = `null`;

exports[`containers/ScrollToTop updates on pathname change 1`] = `null`;
